ibc/testing/chain.go

```diff
@@ -120,8 +120,8 @@ func NewTestChain(t *testing.T, coord *ibcgotesting.Coordinator, chainID string)
 	return chain
 }
 
-func NewTransferPath(chainA, chainB *ibcgotesting.TestChain) *ibcgotesting.Path {
-	path := ibcgotesting.NewPath(chainA, chainB)
```

Comment:
curious why we're importing the type?

Reply:
The root of the issue that breaks the test is the [`SignAndDeliver`](https://github.com/cosmos/ibc-go/blob/d34cef7e075dda1a24a0a3e9b6d3eff406cc606c/testing/simapp/test_helpers.go#L332-L364) function which has the fee hardcoded to `0stake`. This function is called by the function [`SendMsgs`](https://github.com/cosmos/ibc-go/blob/b2fb1192245134607e0e7a294d09d2aa1145f017/testing/chain.go#L313-L347) from the chain type. The chain's `SendMsgs` function is called everywhere on the `Endpoint` type functions, and the `Path` type [has `Endpoint` instances](https://github.com/cosmos/ibc-go/blob/b2fb1192245134607e0e7a294d09d2aa1145f017/testing/path.go#L11-L14). So, to make it work, I had to write a customization of the SignAndDeliver and SendMsgs (avoiding importing this type too) but we needed to import the `Path` and `Endpoint` types to be able to use this new logic